instrumentation-core/servlet-2.2/src/main/java/io/opentelemetry/instrumentation/servlet/ServletHttpServerTracer.java

```diff
@@ -27,7 +28,14 @@
   private static final Logger log = LoggerFactory.getLogger(ServletHttpServerTracer.class);
 
   public Context startSpan(HttpServletRequest request, String spanName) {
-    return startSpan(request, request, request, spanName);
+    Context context = startSpan(request, request, request, spanName);
+
+    SpanContext spanContext = Span.fromContext(context).getSpanContext();
+    // we do this e.g. so that servlet containers can use these values in their access logs
+    request.setAttribute("traceId", spanContext.getTraceId());
+    request.setAttribute("spanId", spanContext.getSpanId());
```

Comment:
I'll send a follow-up PR to change these attribute names to `trace_id` and `span_id` to match logging MDC attribute names.